### PR TITLE
[Build] Define _USE_MATH_DEFINES for FlashMLA targets

### LINKED ISSUE #53935: [Bug]: M_LOG2E availability depends on build configuration in the FlashMLA extension
### Environment

- Windows Server 2025
- Visual Studio Build Tools with MSVC
- CUDA/NVCC 13.2
- Python 3.12
- PyTorch 2.11.0+cu130
- SystemPanic/vllm-windows 0.25-based deployment
- DeepSeek-V4-Flash-0731
- 4 x H200 NVL GPUs

### Background

I encountered this issue while building the vllm-project/FlashMLA fork through SystemPanic/vllm-windows for a deployment on H200 NVL GPUs.

### Problem

The vLLM-specific FlashMLA extension uses `M_LOG2E` in:

    csrc/extension/sm90/dense_fp8/softmax.h
    csrc/extension/sm90/dense_fp8/pybind.cpp

For example:

    const float max_scaled = max(mi) == -INFINITY ? 0.f
                             : max(mi) * (
                                   Scale_max ? scale : float(M_LOG2E));

`M_LOG2E` represents `log2(e)` and is commonly available in GNU/Linux build environments. Its availability is nevertheless implementation-specific.

With MSVC, the `M_*` constants are normally exposed only when `_USE_MATH_DEFINES` is defined before the relevant math header is included. The current source can therefore depend on build flags and translation-unit include order when compiled with NVCC and MSVC.

I understand that native Windows with NVCC and MSVC may not be part of the supported FlashMLA build matrix. However, removing this dependency would be a small cross-platform portability improvement and would not change the existing Linux result.

### Tested source-level fix

The FlashMLA extension built successfully after replacing `float(M_LOG2E)` with the equivalent float literal:

    - float(M_LOG2E)
    + 1.4426950408889634f

For the primary file, the tested change was:

    diff --git a/csrc/extension/sm90/dense_fp8/softmax.h b/csrc/extension/sm90/dense_fp8/softmax.h
    --- a/csrc/extension/sm90/dense_fp8/softmax.h
    +++ b/csrc/extension/sm90/dense_fp8/softmax.h
    @@
    -const float max_scaled = max(mi) == -INFINITY ? 0.f : max(mi) * (Scale_max ? scale : float(M_LOG2E));
    +const float max_scaled = max(mi) == -INFINITY ? 0.f : max(mi) * (Scale_max ? scale : 1.4426950408889634f);

The original expression explicitly converts `M_LOG2E` to a 32-bit `float`. The replacement literal is also a `float` because of the `f` suffix.

Both expressions round to the same IEEE-754 float32 value, so the tested change preserves the numerical result and does not alter the algorithm.

A project-owned named constant could avoid repeating the literal, for example:

    inline constexpr float kLog2E = 1.4426950408889634f;

### Alternative build-level workaround

The issue can also be avoided in the vLLM CMake integration by defining `_USE_MATH_DEFINES` for both FlashMLA extension targets:

    target_compile_definitions(
        _flashmla_C PRIVATE _USE_MATH_DEFINES=1
    )

    target_compile_definitions(
        _flashmla_extension_C PRIVATE _USE_MATH_DEFINES=1
    )

This is a smaller integration-specific workaround and keeps the existing source unchanged.

### Related shared-source occurrences

The same dependency on `M_LOG2E` also appears in source inherited from `deepseek-ai/FlashMLA`, including:

    csrc/api/dense_decode.h
    csrc/sm90/decode/dense/splitkv_mla.cuh
    csrc/sm90/decode/sparse_fp8/splitkv_mla.cuh
    csrc/smxx/decode/combine/combine.cu
    csrc/sm100/prefill/dense/kernel/sm100_fmha_bwd_kernel_tma_warpspecialized.hpp
    csrc/sm100/prefill/dense/kernel/sm100_fmha_bwd_mla_kernel_tma_warpspecialized.hpp

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by arcusbuilds**
## Purpose

Fixes #53935.

FlashMLA uses `M_LOG2E` in the vLLM extension (`csrc/extension/sm90/dense_fp8/{softmax.h,pybind.cpp}`) and in four upstream files that vLLM compiles into `_flashmla_C` (`api/dense_decode.h`, `smxx/decode/combine/combine.cu`, `sm90/decode/{dense,sparse_fp8}/splitkv_mla.cuh`). MSVC only defines `M_*` when `_USE_MATH_DEFINES` is set before `<cmath>` is included, so both targets fail to build with NVCC+MSVC.

This adds `_USE_MATH_DEFINES` to both targets' compile definitions, guarded by `if(MSVC)` so compile flags on other platforms are untouched. One change covers all six files, and it matches what FlashMLA's own `setup.py` already passes to nvcc. Replacing `M_LOG2E` with a literal in the two extension files (the other option in the issue) would leave `_flashmla_C` broken.

## Test Plan

Minimal MSVC translation unit mirroring `softmax.h` (`#include <cmath>` then `float(M_LOG2E)`), built with and without the define. Linux behaviour is unchanged; existing CI covers `tests/kernels/attention/test_flashmla.py`.

## Test Result

```
> cl /nologo /c /EHsc m_log2e_check.cpp
m_log2e_check.cpp(3): error C2065: 'M_LOG2E': undeclared identifier

> cl /nologo /c /EHsc /D_USE_MATH_DEFINES m_log2e_check.cpp
(exit 0)
```

### CODE REVIEW

File: cmake/external_projects/flashmla.cmake [hunk 1]
```diff
@@ -162,8 +162,11 @@ if(FLASH_MLA_ARCHS)
         $<$<COMPILE_LANGUAGE:CUDA>:-std=c++20>)
 
     # _flashmla_C is now ABI-stable torch 2.11+
+    # _USE_MATH_DEFINES: MSVC only exposes M_LOG2E (used throughout FlashMLA's
```
Comment (Harry-Chen) [inline cmake/external_projects/flashmla.cmake]: Please set this only for MSVC (and leave some explanation). Otherwise, it invalidates all our build caches and will have potential side effects.

File: cmake/external_projects/flashmla.cmake [hunk 2]
```diff
@@ -181,7 +184,8 @@ if(FLASH_MLA_ARCHS)
 
     # _flashmla_extension_C is now ABI-stable w/ torch 2.11+
     target_compile_definitions(_flashmla_extension_C PRIVATE
-        TORCH_TARGET_VERSION=0x020B000000000000ULL)
+        TORCH_TARGET_VERSION=0x020B000000000000ULL
+        _USE_MATH_DEFINES)
```
Comment (Harry-Chen) [inline cmake/external_projects/flashmla.cmake]: Same as above.

**Review by Harry-Chen — CHANGES_REQUESTED** [review]

File: cmake/external_projects/flashmla.cmake (hunk 1 above)
Comment (arcusbuilds) [reply cmake/external_projects/flashmla.cmake]: Done, moved the define into a single `if(MSVC)` block covering both targets, with a comment explaining why. Flags on non-MSVC builds are now unchanged.

File: cmake/external_projects/flashmla.cmake (hunk 2 above)
Comment (arcusbuilds) [reply cmake/external_projects/flashmla.cmake]: done!

**Review by Harry-Chen — APPROVED** [review]
Thanks!

**Review by ywang96 — APPROVED** [review]
